app/models/TraderProfile.scala

```diff
@@ -80,4 +81,27 @@ object TraderProfile {
         }
       case Left(errors) => Left(errors)
     }
+
+  def validateHasNirms(
+    answers: UserAnswers
+  ): EitherNec[ValidationError, Boolean] =
+    answers.getPageValue(HasNirmsUpdatePage) match {
+      case Right(true)  => Left(NonEmptyChain.one(UnexpectedPage(HasNirmsUpdatePage)))
+      case Right(false) =>
+        answers.getPageValue(TraderProfileQuery) match {
+          case Right(userProfile) =>
+            if (userProfile.nirmsNumber.isDefined) {
+              answers.getPageValue(RemoveNirmsPage) match {
+                case Right(true)  => Right(true)
+                case Right(false) =>
+                  Left(NonEmptyChain.one(UnexpectedPage(HasNirmsUpdatePage)))
+              }
+            } else {
+              Right(false)
+            }
+          case Left(_)            => Left(NonEmptyChain.one(UnexpectedPage(HasNirmsUpdatePage)))
```

Comment:
i would just pass on the errors as i think it shouldbe page missing not unexpected page